fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/ExistsApplyToJoinTest.java

```diff
@@ -0,0 +1,121 @@
+        EqualTo equalTo = new EqualTo(leftSlots.get(0), rightSlots.get(0));
+        Exists exists = new Exists(right, false);
+        LogicalApply<LogicalOlapScan, LogicalOlapScan> apply =
+                new LogicalApply<>(ImmutableList.of(leftSlots.get(0), rightSlots.get(0)),
+                        exists, Optional.of(equalTo), Optional.empty(),
+                        false, false, left, right);
+        PlanChecker.from(MemoTestUtils.createConnectContext(), apply)
+                .applyTopDown(new ExistsApplyToJoin())
+                .matchesFromRoot(logicalJoin(
+                        logicalOlapScan(),
+                        logicalOlapScan()
+                ).when(j -> j.getJoinType().equals(JoinType.LEFT_SEMI_JOIN)));
+    }
+
+    @Test
+    public void testUnCorrelatedExists() {
+        LogicalOlapScan left = PlanConstructor.newLogicalOlapScan(0, "t1", 1);
+        List<Slot> leftSlots = left.getOutput();
+        LogicalOlapScan right = PlanConstructor.newLogicalOlapScan(0, "t2", 1);
+        List<Slot> rightSlots = right.getOutput();
+        EqualTo equalTo = new EqualTo(leftSlots.get(0), rightSlots.get(0));
+        Exists exists = new Exists(right, false);
+        LogicalApply<LogicalOlapScan, LogicalOlapScan> apply =
+                new LogicalApply<>(Collections.emptyList(),
+                        exists, Optional.of(equalTo), Optional.empty(),
+                        false, false, left, right);
+        PlanChecker.from(MemoTestUtils.createConnectContext(), apply)
+                .applyTopDown(new ExistsApplyToJoin())
+                .matchesFromRoot(logicalJoin(
+                        logicalOlapScan(),
+                        logicalLimit(logicalOlapScan())
+                ).when(j -> j.getJoinType().equals(JoinType.CROSS_JOIN)));
+    }
+
+    @Test
+    public void testNotUnCorrelatedExists() {
+        LogicalOlapScan left = PlanConstructor.newLogicalOlapScan(0, "t1", 1);
+        List<Slot> leftSlots = left.getOutput();
+        LogicalOlapScan right = PlanConstructor.newLogicalOlapScan(0, "t2", 1);
+        List<Slot> rightSlots = right.getOutput();
+        EqualTo equalTo = new EqualTo(leftSlots.get(0), rightSlots.get(0));
+        Exists exists = new Exists(right, true);
+        LogicalApply<LogicalOlapScan, LogicalOlapScan> apply =
+                new LogicalApply<>(Collections.emptyList(),
+                        exists, Optional.of(equalTo), Optional.empty(),
+                        false, false, left, right);
+        PlanChecker.from(MemoTestUtils.createConnectContext(), apply)
+                .applyTopDown(new ExistsApplyToJoin())
+                .matchesFromRoot(logicalFilter(logicalJoin(
+                        logicalOlapScan(),
+                        logicalAggregate(logicalLimit(logicalOlapScan()))).when(
+                            j -> j.getJoinType().equals(JoinType.CROSS_JOIN)))
+                );
+    }
+
+    @Test
+    public void testNotCorrelatedExists() {
```

Comment:
suggest to rename to testCorrelatedExistsToAntiJoin()
the "NotCorrelated" is missleading. "Not" is not use to decrate "Correlated" here